Feat/filter users

GnesisVSS opened the pull request.

GnesisVSS committed:
feat: Added dynamic filter function for user search

Added a filter to search users by email, name and country.

GnesisVSS committed:
fix: fixed the update function

Make request parameters optional and use query parameter for 'rut' instead of route parameter

GnesisVSS committed:
refactor: Fixed http status in login

Changed the http status from default 201 to 200

LuchoSerruya commented on src/user/user.controller.ts:
```diff
@@ -68,9 +70,31 @@ export class UserController {
+    try {
+      return this.userService.update(rut, updateUserDto);
+    } catch (error) {
+      throw new BadRequestException("Error updating user")
```
UserService already throws an exception if something goes wrong - you don't need to catch it and throw a new one in here

LuchoSerruya commented on src/user/user.controller.ts:
```diff
@@ -68,9 +70,31 @@ export class UserController {
+        data: { users },
+      };
+    } catch (error) {
+      throw new BadRequestException('Error searching users');
```
UserService already throws an exception if something goes wrong - you don't need to catch it and throw a new one in here

LuchoSerruya reviewed (changes requested):
Pending changes

GnesisVSS committed:
fix: Fixed redundant exception throwing in login, update, and search

deleted redundant exceptions in user controller for login, update and search functions that were already declarated in user service

GnesisVSS commented:
Fixed requested changes.
